include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -289,7 +289,7 @@ namespace LinearAlgebra
       //  - Third case: the vectors have different size.
       if (vector->getMap()->isSameAs(*V.vector->getMap()))
         {
-          *vector = *V.vector;
+          *vector = Tpetra::createCopy(*V.vector);
```

Comment:
Is this really what you want? This creates a deep copy of `v.vector` and assigns it to `vector`. It involves releasing the previously held memory and allocating some new memory. But we know that the layout of the two vectors is the same. We should simply be copying the elements from `v.vector`'s memory into `vector`'s memory.

Reply:
That's a good point!
I've updated this part. Now the elements from `v.vector` are copied into `vector` via Kokkos views.